### ISSUE TITLE
K0 limit calculation

### ISSUE THREAD

**Issue by ckolbPTB** [PTB-MR/mrpro#50]
I think the calculation of the k0 limits here are wrong:
https://github.com/ckolbPTB/mrpro/blob/52d469d7ebecc7fe5fd8cf352da8a3a0ad17241c/src/mrpro/data/_KData.py#L102

They should be calculated based on `AcqInfo.number_of_samples` and `AcqInfo.center_sample`

**Comment by fzimmermann89** [comment]
Ops, mea culpa. That's an error.

Also: "Center and encoding limits for the readout dimension is not given in the XML header. This is to accommodate sequences where the center of the readout may change from readout to readout (alternating directions of readout). There is a field on the individual data headers to indicate the center of the readout." 
(https://ismrmrd.github.io/apidocs/1.5.0/)

Maybe we should not try to set it at all?
Or only if center and number of samples is the same for each? Or do we want to make the min, center, max of an EncodingLimit potentially a Tensor?

**Comment by fzimmermann89** [comment]
Suggestion:
If it is the same for all samples, we set it to a scalar value
If it is not the same for all samples, we do not set it.

In https://github.com/ckolbPTB/mrpro/blob/478afd74923eda4dc55611a58960ab013560dbbb/src/mrpro/data/traj_calculators/_KTrajectoryRpe.py#L129  and similar, we check if k0 encoding limits has been set, or error if it is None.

So we only have to do the unique logic in one place, and just check if k0 limits is None whenever we use it?
If a Trajectory works with changing k0 encoding limits, it has to look at center_sample and do the appropriate calculation.
(I assume the only trajectory where this would be supported by us might be Cartesian, correct?)

**Comment by ckolbPTB** [comment]
I would suggest to not use it at all or does it contain any information which is not in `number_of_samples` and `center_sample`? 

I think it is - or at least can be - different as soon as you have multi-echo acquisitions with bi-polar gradients independent if they are Cartesian or radial acquisitions.

**Comment by fzimmermann89** [comment]
So you suggestion would beto NEVER set k0 encoding limits?

The only reasons I can see to set k0 is 
1) not to repeat the 
```
if len(unique(number_of_samples))==1 && len(unique(center_sample))==1
#take first values
else:
 error
```
in many places. 

2) ismrmrd kspace_encoding_step_0 can be set for the encoding limits, it is just optional.

If we set it only if it is available in the ismrmrd header, it is easy to have bugs if somebody works with ismrmrd files that set this value, trusts encodinglimits.k0 to be set, but wants to use it for other files that do not set this value.
I would prefer to not have k0's existence dependent on the origin of the data. 

So I'd either 
- never set it, or 
- always set it and fill it in from number_of_samples and center_sample if it is missing in the ismrmrd data.

Question if we always set it: Will most of our trajectories support different center_samples?
- If yes, would always make k0 a tensor, either with one element (broadcastable to number of acsqs) set from the header or usiqing unique on  number_of_samples/center_sample  or a tensor with number of acqs-length calculated from those values for each acq.
- If no, I would set it to 'None' in the not-all-equal case and the trajectorycalculator can then look into the center_sample property to calculate it

**Comment by ckolbPTB** [comment]
Solved in #222:

We always set it if it has not been defined before. Different values for center_sample is supported. Different number of number_of_samples will result in an error when trying to combine acquisitions to a tensor.

The encodig limits should describe the encoded space of all readout lines. If we have two readouts with (number_of_samples, center_sample) of (100, 20) (e.g. partial Fourier in the negative k0 direction) and (100, 80) (e.g. partial Fourier in the positive k0 direction) then this should lead to encoding limits of [min=0, max=159, center=80]. Therefore limit.k0.min is always 0, limit.k0.center is defined by largest center sample (in this case 80) and limit.k0.max is defined by the largest center sample (in this case 80) + the largest value of (number_of_sampled - center_sample - 1) (in this case 80 - 1 = 79).
